.github/workflows/ci-cd.yml

```
@@ -317,9 +333,8 @@ jobs:
   pre-deploy:
     name: Pre-Deploy
     runs-on: ubuntu-latest
-    needs: check
-    # Run only on pushing a tag
-    if: github.event_name == 'push' && contains(github.ref, 'refs/tags/')
+    needs: [pre-setup, check]
```

Let's keep the lists multiline. Plus add context for why unobvious jobs are included explicitly:
```suggestion
    needs:
    - check
    - pre-setup  # transitive, for accessing settings
```